proposals/mutation-support-cli.md

```diff
@@ -0,0 +1,325 @@
+
+```bash
+kyverno apply policy-with-mutate-existing.yaml --target-resource=/path/to/resources 
+```
+
+When this flag is passed but the policy or policies have no mutate existing, the behavior should be to ignore the flag.
+
+When this flag is not passed but the policy has mutate existing rules, it should not error. but should generate a warning that no resources have been passed. and the rule evaluation should be **SKIPPED**.
+
+The target resource flag being passed here maps to the unchanged resource, the apply command will output the mutated resource if succeded
+
+
+### test
+
+three extra keys will be added to the test yaml definition to define the behavior of mutate existing with the test command
+
+```yaml
+apiVersion: cli.kyverno.io/v1alpha1
+kind: Test
+metadata:
+  name: kyverno-test
+policies:
+  - <path/to/policy.yaml>
+  - <path/to/policy.yaml>
+resources:
+  - <path/to/resource.yaml>
+  - <path/to/resource.yaml>
+exceptions:
+  - <path/to/exception.yaml>
+  - <path/to/exception.yaml>
+targetResources: # THIS WILL BE ADDED
+  - <path/to/unmutated/resource.yaml>
+variables: variables.yaml
+mutateExistingTargetVariables: mutate-vars.yaml  # THIS WILL BE ADDED
+userinfo: user_info.yaml
+results: 
+- policy: <name> 
+  isValidatingAdmissionPolicy: false
+  rule: <name>
+  resources:
+  - <namespace_1/name_1>
+  - <namespace_2/name_2>
+  patchedResource: <file_name.yaml>
+  generatedResource: <file_name.yaml>
+  patchedTargetResources: ## THIS WILL BE ADDED
+  - <path/to/patched/resource1.yaml>
+  - <path/to/patched/resource2.yaml>
+  cloneSourceResource: <file_name.yaml>
+  kind: <kind>
+  result: pass
+```
+
+These keys will be optional, all of them will be ignored (whatever value being put to them will not be accounted for) if the policy passed has no mutate existing
+
+The `mutateExistingTargetVariables` key will contain the variables file that will be supplied to replace place holders in target resource selection. e.g: targets that will be selected on specific operations (update, delete, etc.. ) or targets that will be selected based on the role of who performed the admission operation. and its structure and keys will match that of the `variables` key used to define the variables of policies
+
+## Technical details
+
+The CLI isn't going to mutate actual resources, its only required to output the changes that will be made by a mutation rather than the original resource in case of a mutation
+
+### Example
+
+This example adds the label `foo=bar` to deployments in the `secure` namespace 
+
+```yaml
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: policy
+spec:
+  mutateExistingOnPolicyUpdate: true
+  rules:
+    - name: deployment-update
+      match:
+        any: 
+        - resources:
+            kinds:
+            - Deployment
+            namespaces:
+              - "secure"
+      mutate:
+        targets:
+        - apiVersion: apps/v1
+          kind: Deployment
+          namespace: "secure"
```

Comment:
Let's use a different example to illustrate the mutate existing policy, i.e., we can choose to mutate an existing configmap upon namespace creation. The current one can be achieved via the standard admission review mutation.